### PR TITLE
[Branch-2.11] Revert [improve][broker] Require authRole is proxyRole to set originalPrincipal(#19455)

### PR DESCRIPTION
**Pull request by hangc0276**
This reverts commit c7eabc9391361508fc222469479bfea83d5d0c64.

### Motivation
PR #19455 introduced break change and has been cherry-picked to release branch-2.11. It will lead to users upgrading to the new release version authenticate failed.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by nodece — APPROVED** [review]
LGTM

**Review by Technoboy- — APPROVED** [review]

**Comment by hangc0276** [comment]
We need to revert more commits related to this PR.

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java [hunk 1]
```diff
@@ -292,53 +293,19 @@ public CompletableFuture<Boolean> allowSinkOpsAsync(NamespaceName namespaceName,
...
+                        + "request is via proxy.");
+            }
+            if (proxyRoles.contains(originalPrincipal)) {
+                log.warn("Original principal {} cannot be a proxy role ({})", originalPrincipal, proxyRoles);
+                throw new RestException(Response.Status.UNAUTHORIZED, "Original principal cannot be a proxy role");
             }
-        } else if (StringUtils.isNotBlank(originalPrincipal)
-                && !(allowNonProxyPrincipalsToBeEqual && originalPrincipal.equals(authenticatedPrincipal))) {
-            errorMsg = "cannot specify originalPrincipal when connecting without valid proxy role.";
-        }
```
Comment (michaeljmarshall) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/AuthorizationService.java:335]: These are the lines that are under question.

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
> We need to revert more commits related to this PR.

Even if the PMC decides that the right decision is to "revert this behavior change", we should not revert this whole commit. Only a few lines of code are actually under dispute.

**Comment by codelipenghui** [comment]
We have an acceptable solution for now. So that we don't need to revert this change, just push a fix to the release branches.
https://lists.apache.org/thread/q1k1krfjr8njdf8t3sy1ht4vg67s4x71 the discussion thread.

And @michaeljmarshall message me in Slack, he will open PR to fix it.

Thanks @hangc0276 and @michaeljmarshall

**Comment by michaeljmarshall** [comment]
https://github.com/apache/pulsar/pull/19830 should supersede this PR, so I am going to close this for now.

**Pull request closed without merging**